tuf/api/serialization/json.py

```diff
@@ -44,22 +46,39 @@ def deserialize(self, raw_data: bytes) -> Metadata:
 class JSONSerializer(MetadataSerializer):
     """Provides Metadata to JSON serialize method.
 
-    Attributes:
+    Args:
         compact: A boolean indicating if the JSON bytes generated in
-                'serialize' should be compact by excluding whitespace.
+            'serialize' should be compact by excluding whitespace.
+        validate: Check that the metadata object can be deserialized again
+            without change of contents and thus find common mistakes.
+            This validation might slow down serialization significantly.
 
     """
 
-    def __init__(self, compact: bool = False):
+    def __init__(self, compact: bool = False, validate: Optional[bool] = False):
         self.compact = compact
+        self.validate = validate
 
     def serialize(self, metadata_obj: Metadata) -> bytes:
         """Serialize Metadata object into utf-8 encoded JSON bytes."""
+
         try:
+            md_dict = metadata_obj.to_dict()
+
+            if self.validate:
+                try:
+                    new_md_obj = Metadata.from_dict(copy.deepcopy(md_dict))
```

Comment:
is this the correct place to hook in? Now we go from Metadata to "intermediate dictionary format" to Metadata: so we never actually serialize to json during the validation.

We could serialize all the way to json, then do `JSONDeserializer().deserialize()` to go back to metadata maybe?